daft/dataframe/dataframe.py

```diff
@@ -729,6 +729,106 @@ def write_deltalake(
 
         return with_operations
 
+    @DataframePublicAPI
+    def write_lance(
+        self,
+        uri: Union[str, pathlib.Path],
+        mode: Literal["create", "append", "overwrite"] = "create",
+        io_config: Optional[IOConfig] = None,
+    ) -> "DataFrame":
+        """
+        Writes the DataFrame to a Lance table
+        Note:
+            `write_lance` requires python 3.9 or higher
+        Args:
+          uri: The URI of the Lance table to write to
+          mode: The write mode. One of "create", "append", or "overwrite"
+          io_config (IOConfig, optional): configurations to use when interacting with remote storage.
+        Example:
+        --------
+
+
+        >>> df = daft.from_pydict({"a": [1, 2, 3, 4]})
+        >>> df.write_lance("/tmp/lance/my_table.lance")
+        >>> daft.read_lance("/tmp/my_table.lance")
```

Comment:
Noticed these don't match up
```suggestion
        >>> df.write_lance("/tmp/lance/my_table.lance")
        >>> daft.read_lance("/tmp/lance/my_table.lance")
```